src/Nethermind/Nethermind.Core/Crypto/Keccak.cs

```diff
@@ -182,7 +202,7 @@ private KeccakKey(byte[] bytes)
             Bytes = bytes;
         }
 
-        public static implicit operator KeccakKey(Keccak k) => new(k.Bytes);
+        public static implicit operator KeccakKey(Keccak k) => new(k.Bytes.ToArray());
```

Comment:
`KeccakKey` is just a struct wrapper for dictionary keys to devirtualise the dictionary. Can either drop it and switch to `ValueKeccak` or have it backed by `Keccak` rather than `byte[]`; rather than using `.ToArray()`